### PR TITLE
Texlive 2019

### PR DESCRIPTION
**Pull request by extrowerk**
WIP

### CODE REVIEW

**Comment by extrowerk** [comment]
-lbsd and -lnetwork is just a workaround currently.

File: app-text/texlive/texlive-2019.recipe [hunk 1]
```diff
@@ -7,20 +7,20 @@ HOMEPAGE="http://tug.org/texlive/"
...
+SOURCE_URI="ftp://tug.org/historic/systems/texlive/$portVersion/texlive-20190410-source.tar.xz"
+SOURCE_URI_2="ftp://tug.org/historic/systems/texlive/$portVersion/texlive-20190410-texmf.tar.xz"
+CHECKSUM_SHA256="d2a29fef04e34dc3d2d2296c18995fc357aa7625e7a6bbf40fb92d83d3d0d7b5"
+CHECKSUM_SHA256_2="c2ec974abc98b91995969e7871a0b56dbc80dd8508113ffcff6923e912c4c402"
+SOURCE_DIR="texlive-20190410-source"
+SOURCE_DIR_2="texlive-20190410-texmf"
+PATCHES="texlive-2019.patchset"
 ADDITIONAL_FILES="tlutils.pm.patch"
 
 ARCHITECTURES="x86 !x86_gcc2 x86_64"
```
Comment (return) [inline app-text/texlive/texlive-2019.recipe]: I don't think texlive builds on x86_64 yet. But we should probably disable this unless you plan on doing this later in this PR. But your choice if you decide to do this later.

**Comment by return** [comment]
@extrowerk Judging from the updated patches, the build only works on x86 and doesn't work on x86_64 (I tested building this for x86_64).

I don't have a 32 bit build to confirm if it packages fine but the change to make it work on x86_64 can be done later or in this PR since TeX Live is very difficult to package properly.

File: app-text/texlive/texlive-2019.recipe (hunk 1 above)
Comment (extrowerk) [reply app-text/texlive/texlive-2019.recipe]: I can't say it for sure. I checked the files where i386-haiku was added (see patches), but it doesn't mentions any x86_64 platforms. Maybe they are defined in some other file which needs to be patched aswell. 
I am still working on this, so far it built everything (?), but the install phase bails out telling me there is no binary for this platform.

**Comment by Begasus** [comment]
Could check later and let the build run for tonight on gcc2h (still have the download for it)

**Comment by extrowerk** [comment]
The compilation doesnt need much time, but extracting the 10G small files. At least it was slow for me.

**Comment by Begasus** [comment]
At least is was faster then building Qt5 (which has a smaller source base) ;)

**Comment by Begasus** [comment]
running package-specific postactions
Can't open perl script "/boot/system/data/texlive/tlpkg/tlpostcode/dvipdfmx.pl": No such file or directory
./install-tl: calling post action script tlpkg/tlpostcode/dvipdfmx.pl did not succeed in /sources/Master, status 3 at tlpkg/TeXLive/TLUtils.pm line 1699.
Can't open perl script "/boot/system/data/texlive/tlpkg/tlpostcode/ptex2pdf-tlpost.pl": No such file or directory
./install-tl: calling post action script tlpkg/tlpostcode/ptex2pdf-tlpost.pl did not succeed in /sources/Master, status 3 at tlpkg/TeXLive/TLUtils.pm line 1699.
Can't open perl script "/boot/system/data/texlive/tlpkg/tlpostcode/xetex.pl": No such file or directory
./install-tl: calling post action script tlpkg/tlpostcode/xetex.pl did not succeed in /sources/Master, status 3 at tlpkg/TeXLive/TLUtils.pm line 1699.
finished with package-specific postactions

./install-tl: errors in installation reported above

**Comment by Begasus** [comment]
Maybe some sed magic there? Files are actually in ./self/data/...

**Comment by extrowerk** [comment]
@Begasus : thanks for the feedback, it is really useful.
During this week i will have no time to mess with this, but let me try to fix it next week.

**Comment by Begasus** [comment]
@extrowerk No problem, at least we know the build itself is OK :+1:

**Comment by extrowerk** [comment]
I would not say that for sure, at least not for 64 bit, but i try my best to fix it for both platform. Hope you can test my changes later on 32 bit.

**Comment by Begasus** [comment]
@extrowerk any changes worth checking?

**Comment by extrowerk** [comment]
No. I think i failed with this recipe.

**Comment by Begasus** [comment]
Closing in favour of https://github.com/haikuports/haikuports/pull/5963

**Pull request closed without merging**